src/secret_sharing/xor.rs

```diff
@@ -20,3 +26,155 @@ impl XorReplicated {
         self.right
     }
 }
+
+/// XOR secret share.
+///
+/// Bits are stored in the Little-Endian format. Accessing the first element
+/// like `s[0]` will return the LSB.
+///
+/// ## Example
+/// ```
+/// use raw_ipa::secret_sharing::XorSecretShare;
+///
+/// let s = XorSecretShare::<64>::from(2_u128);
+/// let b0 = s[0];
+/// let b1 = s[1];
+///
+/// assert_eq!(b0, 0);
+/// assert_eq!(b1, 1);
+/// ```
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub struct XorSecretShare<const N: usize>(BitVec<u8, Lsb0>);
+
+// BitVec uses `usize` as its default store type in memory. Using `u8`
+// here allows us to store bits larger than 32 but less than 64 most
+// effectively (i.e., 40-bit = 5 * `u8` blocks).
+
+impl<const N: usize> XorSecretShare<N> {
+    const BITS: usize = N;
+    const SIZE_IN_BYTES: usize = Self::BITS / 8;
+
+    #[allow(dead_code)]
+    #[must_use]
+    pub fn new(v: BitVec<u8, Lsb0>) -> Self {
+        Self(v)
+    }
+
+    #[must_use]
+    pub fn zero() -> Self {
+        Self(BitVec::from_element(0))
+    }
+}
+
+impl<const N: usize> BitXor for XorSecretShare<N> {
+    type Output = Self;
+
+    fn bitxor(mut self, rhs: Self) -> Self::Output {
+        self ^= rhs;
+        self
+    }
+}
+
+impl<const N: usize> BitXorAssign for XorSecretShare<N> {
+    fn bitxor_assign(&mut self, rhs: Self) {
+        *self.0.as_mut_bitslice() ^= rhs.0;
+    }
+}
+
+/// Converts from `u128` to `XorSecretShare`. If you wish to create a share
+/// that holds more than 128 bits, use `XorSecretShare::new()` with `bitvec!`
+/// instead.
+impl<const N: usize, T: Into<u128>> From<T> for XorSecretShare<N> {
+    fn from(v: T) -> Self {
+        Self(BitVec::from_slice(
+            &v.into().to_le_bytes()[0..Self::SIZE_IN_BYTES],
+        ))
+    }
+}
+
+impl<const N: usize> Index<usize> for XorSecretShare<N> {
```

Comment:
you would probably need `Index<u8>` too, given that matchkeys are at most 40 bits long and we iterate over bits using `u8`?

Reply:
Our code make use of `enumerate()` a lot, where we index individual elements via `i: usize`. I think we can avoid redundant `try_into().unwrap()` calls by using `usize` here, unless there's a big drawback.